### ISSUE TITLE
Fetch and cache icons

### ISSUE THREAD

**Issue by momobobe** [ImranR98/Obtainium#611]
Currently only an installed App has its icon displayed, which makes it hard to distinguish all the uninstalled Apps from a long list. Maybe it's better to fetch the icon from sources upon adding in list. This is doable since sources of alternative App stores always have icons displayed, and sources of code repo usually contain the icon as PNG files in locations like `.../src/main/` or `.../src/main/res/mipmap-{$dpi}/`. After fetching, the icons should be stored locally in persistent cache unless there is any change, and uninstalling an App won't remove its icon, so in this way App records with empty icon could get eliminated.

**Comment by tusharchopra07** [comment]
Same issue the icons were not showing of the apps that are not installed 
![Screenshot_2023-10-21-09-42-30-272_dev imranr obtainium](https://github.com/ImranR98/Obtainium/assets/128215171/0580a3f3-cbd4-450e-b26c-5e9ccdce7286)

**Comment by wasu-code** [comment]
**GitLab** api returns project avatar `avatar_url` [example](https://gitlab.com/api/v4/projects/shosetsuorg%2Fshosetsu)

**GitHub** allows setting social media preview (Settings > Social preview ), but it's not returned by api (probably present in html though). 
Alternatively most repos include app icon in README so taking first image (or first with 1:1 aspect ratio) would allow adding icons before installation.

**F-droid** seems to provide `icon`, [example when searching](https://search.f-droid.org/api/search_apps?q=obtainium)

---

In terms of distinguishing installed from not installed: maybe half-transparent icons or in grayscale (?). Because I agree we need some good way to separate them.

**Comment by TechniKris** [comment]
> I think there should be a more obvious way to differentiate installed from not installed apps
Agreed. It's still hard to distinguish at first glance whether an app is installed or not if the icon is cached (even despite being greyed out).
Just to throw in more ideas how to visualize it:
- Color code the version number/"not installed" text. For example: red - not installed, green - update available, white - up-to-date. Make that text also **bold** so it stands out even more.
- Color code the entire background of an app's row.
- Place an icon marking not installed apps in the same place as the update icon for updatable entries.

-----

Although, I'm fine with *any* form of marking apps, as long as we get to have cached icons. It'd help a lot in finding what I'm looking for.
